Code/GraphMol/Depictor/testDepictor.cpp

```diff
@@ -1064,6 +1064,208 @@ void testGithub2027() {
+               .lengthSq();
+  }
+  msd /= static_cast<double>(molMatchVect.size());
+  TEST_ASSERT(msd < 1.0e-4);
+  // try with a pattern larger than the reference molecule
+  auto hugePatt = "CCCCCCCCCCCCCCCCCCCCCCCCCCC"_smarts;
+  raised = false;
+  try {
+    RDDepict::generateDepictionMatching2DStructure(
+        *cycloheptylPyrazole, *indazoleRef, -1, hugePatt.get());
+  } catch (const RDDepict::DepictException &) {
+    raised = true;
+  }
+  TEST_ASSERT(raised);
+  // try with an out of range confId
+  raised = false;
+  try {
+    RDDepict::generateDepictionMatching2DStructure(
+        *cycloheptylPyrazole, *indazoleRef, 1, refPatt.get());
+  } catch (const RDKit::ConformerException &) {
+    raised = true;
+  }
+
+  // test using matchVect directly
+  cycloheptylPyrazole->removeConformer(0);
+  MatchVectType matchVect;
+  for (size_t i = 0; i < molMatchVect.size(); ++i) {
+    matchVect.emplace_back(
+        std::make_pair(refMatchVect.at(i).second, molMatchVect.at(i).second));
+  }
+  RDDepict::generateDepictionMatching2DStructure(*cycloheptylPyrazole,
+                                                 *indazoleRef, matchVect);
+  TEST_ASSERT(cycloheptylPyrazole->getNumConformers() == 1);
+  msd = 0.0;
+  for (const auto &pair : matchVect) {
+    msd += (indazoleRef->getConformer().getAtomPos(pair.first) -
+            cycloheptylPyrazole->getConformer().getAtomPos(pair.second))
+               .lengthSq();
+  }
+  msd /= static_cast<double>(matchVect.size());
+  TEST_ASSERT(msd < 1.0e-4);
+  // try with a matchVect larger than the reference molecule
+  MatchVectType matchVectHuge(matchVect);
+  for (size_t i = 0; i < indazoleRef->getNumAtoms(); ++i) {
+    matchVectHuge.emplace_back(std::make_pair(0, 0));
+  }
+  raised = false;
+  try {
+    RDDepict::generateDepictionMatching2DStructure(*cycloheptylPyrazole,
+                                                   *indazoleRef, matchVectHuge);
+  } catch (const RDDepict::DepictException &) {
+    raised = true;
+  }
+  // try with a matchVect with out of range indices
+  MatchVectType matchVectOutOfRange(matchVect);
+  matchVectOutOfRange.emplace_back(std::make_pair(100, 100));
+  raised = false;
+  try {
+    RDDepict::generateDepictionMatching2DStructure(
+        *cycloheptylPyrazole, *indazoleRef, matchVectOutOfRange);
+  } catch (const RDDepict::DepictException &) {
+    raised = true;
+  }
+  // try with an out of range confId
+  raised = false;
+  try {
+    RDDepict::generateDepictionMatching2DStructure(*cycloheptylPyrazole,
+                                                   *indazoleRef, matchVect, 1);
+  } catch (const RDKit::ConformerException &) {
+    raised = true;
+  }
+  BOOST_LOG(rdInfoLog) << "Finished" << std::endl;
+}
+
+void testGenerate2DDepictionAllowRGroups() {
+  BOOST_LOG(rdInfoLog)
+      << "-----------------------\n Test "
+         "generateDepictionMatching2DStructure with allowRGroups"
+      << std::endl;
+  std::string templateMolblock = R"RES(
```

Comment:
same comment here on constructing the molecule more simply

Reply:
Done.